src/java/main/org/apache/zookeeper/server/NettyServerCnxn.java

```diff
@@ -267,10 +267,17 @@ private boolean checkFourLetterWord(final Channel channel,
     {
         // We take advantage of the limited size of the length to look
         // for cmds. They are all 4-bytes which fits inside of an int
-        String cmd = FourLetterCommands.getCmdMapView().get(len);
-        if (cmd == null) {
+        if (!FourLetterCommands.isKnown(len)) {
             return false;
         }
+
+        // ZOOKEEPER-2693: don't execute 4lw if it's not enabled.
+        String cmd = FourLetterCommands.getCommandString(len);
+        if (!FourLetterCommands.isEnabled(cmd)) {
+            LOG.debug("Command {} is not executed because it is not white listed.", cmd);
+            return true;
```

Comment:
Hi @hanm I have attached a patch in the jira  for your reference. the patch is not complete in itself, merge you changes in that patch to make it complete.